src/WebJobs.Extensions.DurableTask/Listener/OutOfProcOrchestrationShim.cs

```diff
@@ -104,6 +116,149 @@ internal async Task<bool> ScheduleDurableTaskEvents(OrchestrationInvocationResul
             return true;
         }
 
+        /// <summary>
+        /// Invokes a DF API based on the input action object.
+        /// Most APIs will return a Task to be awaited, but others such as SignalEntity do not.
+        /// </summary>
+        /// <param name="action">An OOProc action object representing a DF task.</param>
+        /// <param name="returnfireAndForgetTask">Whether to return a `CompletedTask` to represent fire-and-forget APIs. Defaults to `true`.</param>
+        /// <returns>If the API returns a task, the DF task corresponding to the input action. Else, null.</returns>
+        private Task InvokeAPIFromAction(AsyncAction action, bool returnfireAndForgetTask = true)
+        {
+            // Helper function to obtain a list of tasks for compoundTasks.
+            Func<AsyncAction[], Task[]> mapInvokeAPIFromAction = (actions) =>
+            {
+                Task[] tasks = new Task[actions.Length];
+                for (var i = 0; i < actions.Length; i++)
+                {
+                    tasks[i] = this.InvokeAPIFromAction(actions[i]);
+                }
+
+                return tasks;
+            };
+
+            Task fireAndForgetTask = returnfireAndForgetTask ? Task.CompletedTask : null;
```

Comment:
Instead of selectively returning null vs Task.CompletedTask depending on the schema, consider always returning Task.CompletedTask. I understand the desire to keep the SchemaVersion.Original behavior the same, but you can accomplish the same thing by just changing your null check to a `!= Task.CompletedTask` check. Since Task.CompletedTask is a static singleton, this will filter out the fire and forget events from the `Task.WhenAny()` performed in the legacy algorithm.

In general, without nullable references, I feel a strong aversion to returning Task or null in the same function, because whenever I see `Task` as the return type my instinct is to immediately `await` and not bother null checking.